src/Waives.Http/WaivesClient.cs

```diff
@@ -125,19 +126,41 @@ public WaivesClient Login(string clientId, string clientSecret)
         /// <seealso cref="Document.Delete"/>
         public async Task<Document> CreateDocument(Stream documentSource)
         {
+            if (documentSource == null)
+            {
+                throw new ArgumentNullException(nameof(documentSource));
+            }
+
+            if (!documentSource.CanSeek)
+            {
+                // This Stream implementation from ASP.NET Core buffers streams larger than the specified threshold to
+                // a temporary file, up to the buffer limit. Furthermore, it is a seekable Stream, so we can query the
+                // Stream's length
+                documentSource = new FileBufferingReadStream(
+                    documentSource,
+                    1024 * 1024, /* 1MiB */
+                    30 * 1000 * 1000, /* 30 MB */
+                    Path.GetTempPath);
+
+                await documentSource.ReadAsync(new byte[1], 0, 1).ConfigureAwait(false);
```

Comment:
Discussed with @phildrip directly. The aim was to avoid reading and buffering the whole stream to get its length, but I explained that I'd confirmed the `FileBufferingReadStream` only buffers a single byte during this read.